packages/babel-plugin-transform-inline-consecutive-adds/src/index.js

```diff
@@ -0,0 +1,193 @@
+"use strict";
+
+const COLLAPSERS = [
+  "./object-collapser",
+  "./array-collapser",
+  "./array-property-collapser",
+  "./set-collapser"
+].map((x) => {
+  const Collapser = require(x);
+  return new Collapser();
+});
+
+function getFunctionParent(path, scopeParent) {
+  const parent = path.findParent((p) => p.isFunction());
+  // don"t traverse higher than the function the var is defined in.
+  return parent === scopeParent ? null : parent;
+}
+
+function getFunctionReferences(path, scopeParent, references = new Set()) {
+  for (let func = getFunctionParent(path, scopeParent); func; func = getFunctionParent(func, scopeParent)) {
+    const id = func.node.id;
+    const binding = id && func.scope.getBinding(id.name);
+
+    if (!binding) {
+      continue;
+    }
+
+    binding.referencePaths.forEach((path) => {
+      if (!references.has(path)) {
+        references.add(path);
+        getFunctionReferences(path, scopeParent, references);
+      }
+    });
+  }
+  return references;
+}
+
+function getIdAndFunctionReferences(name, parent) {
+  const binding = parent.scope.getBinding(name);
```

Comment:
So what's the todo here? When new scopes are created, they should make sure the bindings are resolved properly (leaves AST in consistent state).